AAI/AAIConnectProfile.md

```diff
@@ -747,6 +832,123 @@ See the [GA4GH Passport
+
+2.  `ga4gh_passport:self_contained_v1` JSON field: for use by previously allocated
+    Self-Contained Passports.
+
+`targets` only for use by clients providing Passport-Scoped Access Tokens,
+i.e. the client MUST NOT change the audience of an existing Self-Contained Passport.
+If `targets` is present it MUST contain at least one entry:
+
+    -   One Self-Contained Passport will be generated for each entry in the
+        `targets` map, or the Broker MUST return an error if there are too
+        many entries.
+
+    -   `<aud>`: the target audience string for the token to be included
+        as one of the array entries in the `aud` JWT claim as part of the
+        Self-Contained Passport response.
+        
+    -   `<visa-ref>`: the `value` string of a Visa. The access MUST NOT provide
+        more access than the authorization allowed by the token as part of calling
+        this endpoint.
+        
+        -   However, the `<visa-ref>` may offer a more specific subset
+            of data of any visa that is available by Visas within scope of the
+            caller's token.
+            
+        -   The format of `<visa-ref>` for subsetting access within a broader
+            `<visa-ref>` is up to the Visa Issuer, and is beyond the scope of this
+            specification.
+            
+    -   Example usage of requesting multiple tokens would be to federate calls
+        across many nodes in a network, each expecting a different audience.
+        (e.g. federating across 100 GA4GH Beacon nodes from different providers)
+
+#### Broker Passport Endpoint Processing
+
+1.  The Broker MUST validate the authorization token received according to
+    requirements for tokens elsewhere in this specification.
+
+2.  The scope of authorization MUST NOT exceed that of the caller's authorization,
+    whether using an `Authorization` header or a Self-Contained Passport.
+    
+    1.  The Broker SHOULD indicate errors in the response for each `<visa-ref>`
+        that attempts to upscope access, and still provide Self-Contained Passports
+        for any remaining `<visa-ref>` unless there are no valid `<visa-ref>`
+        entries.
+
+3.  The Broker SHOULD limit the number of tokens it can produce to some reasonable
+    number, and indicate such a number as part of OIDC Discovery.
+
+#### Passport Endpoint Response
+
+If the Broker's token limit was not exceeded and at least one valid token was minted,
+the Broker's response takes the following form:
+
+```
+Content-Type: application/json
+
+{
+  "tokens": {
+    "<aud>": "<self-contained-passport-jws>",
+    ...
+  },
+  "status": {
+    "aud": {
+      "<aud>": <status-code>,
+      ...
+    }
+    "visas": {
+      "<visa-ref>": <status-code>,
+      ...
+    }
+  }
+}
+```
+
+Where:
+
+-   `tokens`: REQUIRED as the map of tokens per `aud` in the request structure
+    and each `<self-contained-passport-jws>` is a JWS Compact String of a Self-Contained
```

Comment:
Would need to specify the claims within the <self-contained-passport-jws> and how it is similar or different than what is returned from the /userinfo endpoint. For example, it has the visas but doesn't have email, picture, given name, etc.

Reply:
1. It is designed to remove the need for bearer tokens to call back to /userinfo by issuing embedded tokens. More on requirements in the design doc.
2. It is designed to be the mechanism to be leveraged to support visa downscoping, not yet shown in this PR. However there are other endpoints and mechanisms being explored in the design doc.
3. It may offer a different set of claims than /userinfo to not expose as much PII. Once again, there may be other ways to control this using scopes.

Therefore, it is not clear if this PR currently has the right approach, and so this PR so far is for discussion on what such a self-contained passport minting endpoint might look like by exposing more details of how it translates into spec, but these design issues need to be resolved before creating or updating a PR like this to be ready to land.